### PR TITLE
06 time management solution

### PR DESCRIPTION
**Pull request by SergeyChekanov**
Solution for 6 lesson time management

### CODE REVIEW

File: 06-TimeManagement/time_mod/time_mod.c [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	unsigned long array_stat[ARR];
+} stat;
+
+static ssize_t abs_time_show(struct class *class,
+				struct class_attribute *attr,
+						char *buf)
+{
+	getnstimeofday(&ts);
+	if (one_read_abs) {
+		pr_info("abs time %lu.%lu\n", prev_ts.tv_sec,
```
Comment (yekovalyov) [inline 06-TimeManagement/time_mod/time_mod.c]: Is this format string correct?

File: 06-TimeManagement/time_mod/time_mod.c [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+						prev_ts.tv_nsec);
+		sprintf(buf, "Abs time: %lu.%lu\n", prev_ts.tv_sec,
```
Comment (yekovalyov) [inline 06-TimeManagement/time_mod/time_mod.c]: Is this format string correct?

File: 06-TimeManagement/time_mod/time_mod.c [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/device.h>
+#include <linux/err.h>
+#include <linux/rtc.h>
+#include <linux/time.h>
+
+#define ARR 600
+#define BUF 81
```
Comment (yekovalyov) [inline 06-TimeManagement/time_mod/time_mod.c]: Is 81 enough for all possible value of parsed line?

File: 06-TimeManagement/time_mod/time_mod.c [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+		if (stat.count <= ARR) {
+			stat.count++;
+		}
+	}
+	mod_timer(&calc_timer, jiffies + HZ);
+}
+
+static ssize_t cpu_load_show(struct class *class,
+				struct class_attribute *attr,
+						char *buf)
```
Comment (yekovalyov) [inline 06-TimeManagement/time_mod/time_mod.c]: Is the indentation correct?

File: 06-TimeManagement/time_mod/time_mod.c (hunk 1 above)
Comment (SergeyChekanov) [reply 06-TimeManagement/time_mod/time_mod.c]: Rewrited all task

File: 06-TimeManagement/time_mod/time_mod.c (hunk 2 above)
Comment (SergeyChekanov) [reply 06-TimeManagement/time_mod/time_mod.c]: Rewrited all task

File: 06-TimeManagement/time_mod/time_mod.c (hunk 3 above)
Comment (SergeyChekanov) [reply 06-TimeManagement/time_mod/time_mod.c]: Rewrited all task

File: 06-TimeManagement/time_mod/time_mod.c (hunk 4 above)
Comment (SergeyChekanov) [reply 06-TimeManagement/time_mod/time_mod.c]: Rewrited all task

File: 06-TimeManagement/relation_time/time_mod.c [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+				char *buf)
+{
+	pr_info("mymodule: read data");
+
+	unsigned long diff = jiffies - prevJiffiesVal;
+
+	prevJiffiesVal = jiffies;
+
+	sprintf(buf, "time passed since previous read: %u.%03u seconds\n",
+		diff/1000, diff%1000);
```
Comment (yekovalyov) [inline 06-TimeManagement/relation_time/time_mod.c]: How many jiffies are in one second?

Comment (ChekanovSerhii) [reply 06-TimeManagement/relation_time/time_mod.c]: In my virtualbox kernel 250

Comment (ChekanovSerhii) [reply 06-TimeManagement/relation_time/time_mod.c]: Я проверил на своей рабочей тетради с virtualbox - 250, правильно ли зафиксировать с делением 250?

Comment (SergeyChekanov) [reply 06-TimeManagement/relation_time/time_mod.c]: I checked on my work notebook with virtualbox - 250, correct commit with divide 250? jiffies  depend from build kernel

Comment (yekovalyov) [reply 06-TimeManagement/relation_time/time_mod.c]: It is better to use constant HZ for conversation from jiffies to seconds = jiffies / HZ

Comment (SergeyChekanov) [reply 06-TimeManagement/relation_time/time_mod.c]: Done.

**Pull request merged**